MeshLib/ElementCoordinatesMappingLocal.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#include "ElementCoordinatesMappingLocal.h"
+
+#include <limits>
+#include <cassert>
+
+#include "GeoLib/AnalyticalGeometry.h"
+
+#include "MathLib/MathTools.h"
+#include "MeshLib/Node.h"
+
+namespace MeshLib
+{
+
+ElementCoordinatesMappingLocal::ElementCoordinatesMappingLocal(
+    const Element& e,
+    const CoordinateSystem &global_coords)
+: _coords(global_coords)
+{
+    assert(e.getDimension() <= global_coords.getDimension());
+    for(unsigned i = 0; i < e.getNNodes(); i++)
+        _point_vec.push_back(new MeshLib::Node(*(e.getNode(i))));
+
+    getRotationMatrixToGlobal(e, global_coords, _point_vec, _matR2global);
+    rotateToLocal(e, global_coords, _matR2global.transpose(), _point_vec);
+}
+
+ElementCoordinatesMappingLocal::~ElementCoordinatesMappingLocal()
+{
+	for (auto p : _point_vec) delete p;
+}
+
+void ElementCoordinatesMappingLocal::rotateToLocal(
+    const Element &ele,
+    const CoordinateSystem &global_coords,
+    const RotationMatrix &matR2local,
+    std::vector<MeshLib::Node*> &vec_pt) const
+{
+    // rotate the point coordinates
+    for(unsigned i = 0; i < ele.getNNodes(); i++)
+    {
+        Eigen::Vector3d  x_new = matR2local * Eigen::Map<Eigen::Vector3d>(const_cast<double*>(vec_pt[i]->getCoords()));
```

Comment:
Is there an obstacle to implement something like
```
MeshLib::Node operator*(EigenMatrix const& mat, MeshLib::Node const& node)
```
or
```
MathLib::Point3d operator*(EigenMatrix mat, MathLib::Point3d const& )?
```
I would prefer the second version, which is more general.

This would be 
(1) more readable for the user and
(2) save some lines of code and
(3) omit some copies.